src/Microsoft.Health.Fhir.Shared.Core/Features/Search/SearchOptionsFactory.cs

```diff
@@ -106,6 +106,32 @@ public SearchOptions Create(string compartmentType, string compartmentId, string
                 {
                     // TODO: We need to handle format parameter.
                 }
+                else if (string.Equals(query.Item1, KnownQueryParameterNames.Type, StringComparison.OrdinalIgnoreCase))
+                {
+                    var types = query.Item2.SplitByOrSeparator();
+                    var badTypes = types.Where(type => !ModelInfoProvider.IsKnownResource(type)).ToHashSet();
+
+                    if (badTypes.Count != 0)
+                    {
+                        _contextAccessor.FhirRequestContext.BundleIssues.Add(
+                            new OperationOutcomeIssue(
+                                OperationOutcomeConstants.IssueSeverity.Warning,
+                                OperationOutcomeConstants.IssueType.NotSupported,
+                                string.Format(Core.Resources.InvalidTypeParameter, badTypes.OrderBy(x => x).Select(type => $"'{type}'").JoinByOrSeparator())));
+                        if (badTypes.Count != types.Count)
+                        {
+                            searchParams.Add(KnownQueryParameterNames.Type, types.Except(badTypes).JoinByOrSeparator());
+                        }
+                        else
+                        {
+                            searchParams.Add(KnownQueryParameterNames.Type, query.Item2);
```

Comment:
Could we add a comment for why we are including the bad types if there are only bad types. I know we discussed this but it looks like an area that we will forget what our reasoning was in a few months.